source/module_hsolver/hsolver_pw.cpp

```diff
@@ -547,19 +324,10 @@ template <typename T, typename Device>
 void HSolverPW<T, Device>::updatePsiK(hamilt::Hamilt<T, Device>* pHamilt, psi::Psi<T, Device>& psi, const int ik)
 {
     psi.fix_k(ik);
-    if (GlobalV::psi_initializer) // new psi initialization method branch
-    {
-        // do nothing here, because we have already initialize, allocate and make initial guess
-        // basis_type lcao_in_pw function may be inserted here
-    }
-    else if (!this->initialed_psi) // old psi initialization method branch
-    {
-        if (GlobalV::BASIS_TYPE == "pw")
-        {
-            hamilt::diago_PAO_in_pw_k2(this->ctx, ik, psi, this->wfc_basis, this->pwf, pHamilt);
-        }
-        /* lcao_in_pw now is based on newly implemented psi initializer, so it does not appear here*/
-    }
+    if (!GlobalV::psi_initializer && !this->initialed_psi && GlobalV::BASIS_TYPE == "pw")
+    hamilt::diago_PAO_in_pw_k2(this->ctx, ik, psi, this->wfc_basis, this->pwf, pHamilt);
+    /* lcao_in_pw now is based on newly implemented psi initializer, so it does not appear here*/
```

Comment:
if the psi_initializer can replace the old wavefunc thoroughly, these lines can be deleted?